xds/internal/test/e2e/e2e.go

```diff
@@ -0,0 +1,172 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package e2e implements xds e2e tests using go-control-plane.
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"google.golang.org/grpc"
+	channelzpb "google.golang.org/grpc/channelz/grpc_channelz_v1"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+)
+
+func cmd(path string, logger io.Writer, args []string, env []string) (*exec.Cmd, error) {
+	cmd := exec.Command(path, args...)
+	cmd.Env = append(os.Environ(), env...)
+	cmd.Stdout = logger
+	cmd.Stderr = logger
+	return cmd, nil
+}
+
+const (
+	clientStatsPort = 60363 // TODO: make this different per-test, only needed for parallel tests.
+)
+
+type client struct {
+	cmd *exec.Cmd
+
+	target  string
+	statsCC *grpc.ClientConn
+}
+
+// newClient create a client with
+// - xds bootstrap file at "./configs/bootstrap-<testName>.json" (%s is the test name)
+// - xds:///<testName> as the target
+func newClient(testName string, binaryPath string, bootstrap string, logger io.Writer, flags ...string) (*client, error) {
+	target := fmt.Sprintf("xds:///%s", testName)
```

Comment:
I would prefer just passing in the target here. If we need the `testName` at some point in the future, we can add it. But now this variable name does not convey enough information.

Reply:
That's fair. Done